default.nix

```diff
@@ -1,6 +1,6 @@
 (import (
   let
-    lock = builtins.fromJSON (builtins.readFile ./flake.lock);
+    lock = builtins.fromJSON (builtins.readFile ./flake/dev/flake.lock);
```

Comment:
Why do we need the dev lockfile for flake-compat?

Reply:
It's less that we need it in the dev lockfile, and more that we **don't** need it in the root lockfile.

My thought process for this PR was every input we previously claimed was optional, should be part of the partitioned off submodule's inputs.

For this specific input, we should probably not use flakes at all and just [`fetchTarball`](https://nix.dev/manual/nix/2.18/language/builtins#builtins-fetchTarball) it explicitly in the relevant file(s), but that felt out of scope for this PR.

Reply:
I'd prefer not removing it from the main flake then (for now). I made it a flake input in order to have it updated by CI, we could do another mechanism for that

Reply:
The dev flake is also updated by CI, so I don't see the issue.

However isn't the URL versioned, and therefore not updatable, anyway? So the flake lock is only managing that hash for us, not the version.